avalanche/training/supervised/der.py

```
@@ -0,0 +1,263 @@
+            if target not in cl_idxs:
+                cl_idxs[target] = []
+            cl_idxs[target].append(idx)
+
+        # Make AvalancheSubset per class
+        cl_datasets = {}
+        for c, c_idxs in cl_idxs.items():
+            subset = classification_subset(new_data_with_logits, indices=c_idxs)
+            cl_datasets[c] = subset
+        # Update seen classes
+        self.seen_classes.update(cl_datasets.keys())
+
+        # associate lengths to classes
+        lens = self.get_group_lengths(len(self.seen_classes))
+        class_to_len = {}
+        for class_id, ll in zip(self.seen_classes, lens):
+            class_to_len[class_id] = ll
+
+        # update buffers with new data
+        for class_id, new_data_c in cl_datasets.items():
+            ll = class_to_len[class_id]
+            if class_id in self.buffer_groups:
+                old_buffer_c = self.buffer_groups[class_id]
+                # Here it uses underlying dataset
+                old_buffer_c.update_from_dataset(new_data_c)
+                old_buffer_c.resize(strategy, ll)
+            else:
+                new_buffer = ReservoirSamplingBuffer(ll)
+                new_buffer.update_from_dataset(new_data_c)
+                self.buffer_groups[class_id] = new_buffer
+
+        # resize buffers
+        for class_id, class_buf in self.buffer_groups.items():
+            self.buffer_groups[class_id].resize(
+                strategy, 
+                class_to_len[class_id])
+
+
+class DER(SupervisedTemplate):
+    """ 
+    Implements the DER and the DER++ Strategy, 
+    from the "Dark Experience For General Continual Learning" 
+    paper, Buzzega et. al, https://arxiv.org/abs/2004.07211 
+    """
+
+    def __init__(
+        self,
+        model: Module,
+        optimizer: Optimizer,
+        criterion=CrossEntropyLoss(),
+        mem_size: int = 200,
+        batch_size_mem: int = None,
+        derpp: bool = False,
```

I would remove this argument and suggest the user to set `beta=0`.
Otherwise, if a user sets `beta` but forgets `derpp` it will silently ignore the argument.